doc/developer/platform/ux.md

```diff
@@ -299,17 +299,63 @@ The HTTPS server exposes two APIs:
     additional control over query execution and result fetching, a la the
     [Snowflake SQL REST API].
 
-    TODO: fully describe the output format.
+    ### Output format
 
-    TODO: design support for binding parameters.
+    The output format is a JSON object with one key, `results`, whose values is
+    an array of serialized `adapter::command::SimpleResult`, representing all of
+    the statements' executions.
+
+    The serialized `adapter::command::SimpleResult` values are representing in JSON as:
+
+    `SimpleResult` value | JSON value
+    ---------------------|------------
+    `Rows` | `{"rows": <2D array of JSON-ified results>, "col_names": <array of text>}`
+    `Error` | `{"error": <Error string from execution>}`
+    `Ok` | `{ "ok": <tag>}`
+
+    ### Binding parameters
+
+    You can include parameters alongside the query by including on the query a
+    key `"parameters"` whose value is an array of objects of the form:
+
+    ```json
+    {
+      "type": "<resolvable type name>",
+      "value": "<JSON-ified value>"
+    }
+    ```
+
+    You can then reference these parameters in the query as you would with PG,
+    i.e. using 1-indexed references to the parameters array, prefaced by a `$`. e.g.
+
+    ```json
+    {
+      "sql": "INSERT INTO foo VALUES ($1);",
+      "parameters": [{"type": "int", "value": 1}]
+    }
+    ```
 
   * `ws://<ENDPOINT>/query?query=QUERY` streams the result of a single query
     over the WebSockets protocol.
 
     This API is particularly useful for listening to the results of `TAIL`
     queries from a browser.
 
-    TODO: fully describe the output format.
+    ### Output
+
+    A Websocket request outputs the following:
+      - `{"col_names": <array of text>}` if columns of data will be returned
+      - `{"ok": <tag>}` if the command generates a tag (e.g. not `ExecuteResponse::SendingRows`)
+      - `{"error": <Error string from execution>}`
+      - `{"rows": <2D array of JSON-ified results>}` if columns of data will be returned
```

Comment:
It's unclear what this means for TAIL. Can it return multiple `rows`, each with many rows?

Reply:
Clarified